pandas/tests/groupby/test_groupby.py

```diff
@@ -876,14 +871,45 @@ def test_omit_nuisance(df):
         grouped.agg(lambda x: x.sum(0, numeric_only=False))
 
 
-def test_omit_nuisance_sem(df):
-    # GH 38774 - sem should work with nuisance columns
+@pytest.mark.parametrize(
+    "agg_function",
+    ["max", "min"],
+)
+def test_keep_nuisance_agg(df, agg_function):
+    # GH 38815
     grouped = df.groupby("A")
-    result = grouped.sem()
-    expected = df.loc[:, ["A", "C", "D"]].groupby("A").sem()
+    result = getattr(grouped, agg_function)().columns
+    expected = df.loc[:, ["B", "C", "D"]].columns
+    np.testing.assert_array_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "agg_function",
+    ["sum", "mean", "prod", "std", "var", "median"],
+)
+def test_omit_nuisance_agg(df, agg_function):
+    # GH 38774, GH 38815
+    grouped = df.groupby("A")
+    result = getattr(grouped, agg_function)()
+    expected = getattr(df.loc[:, ["A", "C", "D"]].groupby("A"), agg_function)()
     tm.assert_frame_equal(result, expected)
 
 
+@pytest.mark.parametrize(
+    "agg_function",
+    ["skew"],
+)
+def test_omit_nuisance_warnings(df, agg_function):
+    # GH 38815
+    with tm.assert_produces_warning(
```

Comment:
what is this warning? why are you adding extra options?

Reply:
This was the warning that you wanted to be asserted. I took away the parameterization now

https://github.com/pandas-dev/pandas/pull/43674/checks?check_run_id=3771738681
need sto assert this warning happens (this is pretty new btw)